### PR TITLE
Adjust keybindings for deletion in the project panel

### PR DESCRIPTION
**Pull request by mikayla-maki**
Release Notes:

- Improve compatibility keybindings (Atom, JetBrains, TextMate)

### CODE REVIEW

File: assets/keymaps/default-macos.json [hunk 1]
```diff
@@ -613,8 +613,12 @@
       "enter": "project_panel::Rename",
       "backspace": ["project_panel::Trash", { "skip_prompt": false }],
       "delete": ["project_panel::Trash", { "skip_prompt": false }],
-      "cmd-backspace": ["project_panel::Delete", { "skip_prompt": false }],
-      "cmd-delete": ["project_panel::Delete", { "skip_prompt": false }],
+      "shift-backspace": ["project_panel::Delete", { "skip_prompt": false }],
+      "shift-delete": ["project_panel::Delete", { "skip_prompt": false }],
```
Comment (notpeter) [inline assets/keymaps/default-macos.json:617]: cmd+backspace/delete and cmd+shift+backspace/delete look right.

Is there a reason we're adding shift+backspace and shift+delete? I've never seen them

**Comment by notpeter** [comment]
I've made a few changes, but I feel this matches my expectations:

- MacOS:
  - cmd+backspace == trash without prompt (match finder)
  - cmd+shift+backspace == delete with prompt (match finder)
  - delete == trash with prompt (match vscode)
  - backspace (no-op, previously trash with prompt)
- Linux: Match native file explorers
  - delete == delete with prompt
  - shift+delete == delete with prompt
  - backspace (no-op, previously trash with prompt)

I feel relatively strongly that a backspace key should not delete files. This is not the behavior on Windows, MacOS or Linux (that I am aware) nor in VSCode.  I think it's just too easy to press backspace when you thought your focus was in the editor and not the project panel.

I could be convinced that delete on linux should be `delete with prompt`.

File: assets/keymaps/default-linux.json [hunk 2]
```diff
@@ -587,7 +587,7 @@
       "f2": "project_panel::Rename",
       "enter": "project_panel::Rename",
       "shift-delete": ["project_panel::Delete", { "skip_prompt": false }],
-      "delete": ["project_panel::Trash", { "skip_prompt": true }],
+      "delete": ["project_panel::Trash", { "skip_prompt": false }],
```
Comment (SomeoneToIgnore) [inline assets/keymaps/default-linux.json:590]: To note, this alone will remove all keybindings from mac menu entries.

Comment (notpeter) [reply assets/keymaps/default-linux.json:590]: > To note, this alone will remove all keybindings from mac menu entries.

I don't understand, please say more.

1. Isn't this `-linux.json`?
2. How does adding a single binding to scoped to ProjectPanel remove all keybindings for mac menu entries?

Comment (SomeoneToIgnore) [reply assets/keymaps/default-linux.json:590]: Sorry, I've actually missed that this fixes the issue with Linux keybindings, and my comment is also wrong as was about mac, please disregard it entirely.

1. Yes, sorry.
2. It's not adding, but altering the keybinding's value what happens in the PR. If that gets out of sync with the value, defined in the menu, the binding keys won't be displayed in the UI.

In this case they actually start to match, so all good:
https://github.com/zed-industries/zed/blob/1e0e1c7c3199c412b49cbed56c7ad4087fc5842e/crates/project_panel/src/project_panel.rs#L504

Comment (notpeter) [reply assets/keymaps/default-linux.json:590]: Thank you. This is helpful to know that the menu items and keybinds could be out of sync in the actions they perform.

**Pull request merged**